cmd/compute-domain-kubelet-plugin/computedomain.go

```diff
@@ -266,6 +328,10 @@ func (m *ComputeDomainManager) GetNodeIPs(ctx context.Context, cdUID string) ([]
 		return nil, fmt.Errorf("error getting status of nodes in ComputeDomain: %w", err)
 	}
 
+	if len(cd.Status.Nodes) != cd.Spec.NumNodes {
+		return nil, fmt.Errorf("not all nodes populated in ComputeDomain status yet: %w", err)
+	}
+
 	var ips []string
 	for _, node := range cd.Status.Nodes {
 		if m.cliqueID == node.CliqueID {
```

Comment:
Under which condiitions would we expect a different `CliqueID` here? It seems that we're only setting this value once for the nodes.

Reply:
When the compute domain spans nodes that have GPUs from different cliques. When this happens, we *only* want to include the node IPs in the `node_config.cfg` from nodes that have GPUs with the same cliqueID as the current node.